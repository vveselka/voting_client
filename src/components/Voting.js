import React, { Component } from 'react';
import Winner from './Winner';
import Vote from './Vote';
import { connect } from 'react-redux';
// import * as action_creators from "../action_creators";
import * as actionCreators from '../action_creators';

export class Voting extends Component {
  render() {
    const { winner } = this.props;
    return (
      <div className="voting">
        {winner ? <Winner winner={winner} /> : <Vote {...this.props} />}
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    pair: state.getIn(['vote', 'pair']),
    hasVoted: state.get('hasVoted'),
    winner: state.get('winner')
  };
}
// export const VotingContainer = connect(mapStateToProps)(Voting);
export const VotingContainer = connect(mapStateToProps, actionCreators)(Voting);
