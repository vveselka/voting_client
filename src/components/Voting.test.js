import Voting from './Voting';
import renderer from 'react-test-renderer';
import React from 'react';
import ReactDOM from 'react-dom';

import { shallow, mount, render } from 'enzyme';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Voting />, div);
});

it('renders correctly', () => {
  const div = renderer.create(<Voting pair={['fisrt', 'second']} />).toJSON();
  expect(div).toMatchSnapshot();
});

it('should render without throwing an error', () => {
  expect(shallow(<Voting />).exists(<div className="vote" />)).toBe(true);
});
