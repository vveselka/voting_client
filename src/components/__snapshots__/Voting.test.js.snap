// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="voting"
>
  <div
    className="vote"
  >
    <button
      disabled={false}
      onClick={[Function]}
    >
      <h1>
        fisrt
      </h1>
    </button>
    <button
      disabled={false}
      onClick={[Function]}
    >
      <h1>
        second
      </h1>
    </button>
  </div>
</div>
`;
